organization_token.go

```diff
@@ -28,8 +36,14 @@ type OrganizationTokens interface {
 	// Read an organization token.
 	Read(ctx context.Context, organization string) (*OrganizationToken, error)
 
+	// Read an organization token with options.
+	ReadWithOptions(ctx context.Context, organization string, options OrganizationTokenReadOptions) (*OrganizationToken, error)
+
 	// Delete an organization token.
 	Delete(ctx context.Context, organization string) error
+
+	// Delete an organization token with options.
+	DeleteWithOptions(ctx context.Context, organization string, options OrganizationTokenDeleteOptions) error
```

Comment:
With adding these options, we should have some validations for them with clear errors to surface any invalid token types given.  [You can see here](https://github.com/hashicorp/go-tfe/blob/main/admin_setting_smtp.go) for a general example of how we typically do it.

Reply:
Alrighty, so yes I could add some validation logic there but I'm having some thoughts about should I ...

Is there any danger in a user supplying their own token-type value? Not really, the server will catch it, and does return a useful HTTP error message. Also, it's a nice escape-hatch i.e. when HCP Terraform supports a new token-type, this library doesn't require an immediate update, instead users can supply the new token-type as a string instead of using our pre-defined constant list.

Which does raise a different question ... Is this a helper enumeration (i.e. "Here's a list of common values you can use!") or is it an explicit list that this project will enforce? (i.e. "Thou shalt only use these approved values"). Right now I'm leaning towards the former.

If we did want to go down the enforcement route, wouldn't a properly typed value be better instead of late validation? Or shouldn't validation occur when creating the options, not when they're used?

Reply:
@Maed223 Did you have any thoughts or responses to this?

Reply:
In retrospect this may be a bit of a nit, and the example I provided happens to be a case where those potential values will be unlikely to change. Also with the value is optional, there's not much that could go wrong.